src/fu-remote-list.c

```diff
@@ -658,6 +658,25 @@ fu_remote_list_load_metainfos(XbBuilder *builder, GError **error)
 	return TRUE;
 }
 
+gboolean
+fu_remote_list_toggle_test_remotes(FuRemoteList *self, gboolean enable, GError **error)
+{
+	g_return_val_if_fail(FU_IS_REMOTE_LIST(self), FALSE);
+
+	/* not yet initialized */
+	if (self->silo == NULL)
+		return TRUE;
+
+	self->testing_remote = enable;
```

Comment:
maybe check if `self->testing_remote == enable` and return here?

Reply:
fixed